gmusicapi/clients/musicmanager.py

```diff
@@ -512,9 +512,9 @@ def upload(self, filepaths, transcode_quality='320k', enable_matching=False):
             if sample_res.response_code == upload_pb2.TrackSampleResponse.MATCHED:
                 self.logger.info("matched '%s' to sid %s", path, sample_res.server_track_id)
 
-                if enable_matching:
-                    matched[path] = sample_res.server_track_id
-                else:
+                matched[path] = sample_res.server_track_id
+
+                if not enable_matching:
                     self.logger.exception("'%s' was matched without matching enabled", path)
```

Comment:
this isn't directly related to the change, but I think this ought to be `self.logger.error` instead since we're not in an exception handler.